model/task/task.go

```diff
@@ -931,6 +939,36 @@ func (t *Task) MarkAsHostDispatched(hostId, distroId, agentRevision string,
 	return nil
 }
 
+// MarkAsContainerDispatched marks that the container task has been dispatched
+// to a pod. If the task is part of a display task, the display task is also
+// marked as dispatched to a pod.
+func (t *Task) MarkAsContainerDispatched(ctx context.Context, env evergreen.Environment, agentVersion string, dispatchedAt time.Time) error {
+	query := shouldContainerTaskDispatchQuery()
+	query[StatusKey] = evergreen.TaskContainerAllocated
```

Comment:
Allocated in this case refers to whether the container task has had a dedicated pod requested for it to run and mostly exists to distinguish it from the unallocated state (i.e. waiting for a pod to be requested). In the case of host tasks, hosts are started according to aggregate distro-level statistics on the current task queue state, so a host is never solely allocated to run a single task. Host tasks remain in the undispatched state and stay in the task queue until the agent picks it up, at which point it's dispatched.

The two task status differences can merge once the task is dispatched to an agent since the task lifecycle that involves the agent is the same.